packages/react/src/shared/proxy/utilsProxy.ts

```diff
@@ -197,87 +196,52 @@ export function createReactQueryUtilsProxy<
-            }
-            case 'invalidate': {
-              const [input, ...rest] = args as Parameters<
-                AnyDecoratedProcedure['invalidate']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.invalidateQueries(queryKey, ...rest);
-            }
-            case 'refetch': {
-              const [input, ...rest] = args as Parameters<
-                AnyDecoratedProcedure['refetch']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.refetchQueries(queryKey, ...rest);
-            }
-            case 'cancel': {
-              const [input, ...rest] = args as Parameters<
-                AnyDecoratedProcedure['cancel']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.cancelQuery(queryKey, ...rest);
-            }
-            case 'setData': {
-              const [updater, input, ...rest] = args as Parameters<
-                AnyDecoratedProcedure['setData']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.setQueryData(queryKey, updater, ...rest);
-            }
-            case 'setInfiniteData': {
+          const opts = (name: typeof utilName) => {
+            if (['setData', 'setInfiniteData'].includes(name)) {
               const [updater, input, ...rest] = args as Parameters<
-                AnyDecoratedProcedure['setInfiniteData']
+                AnyDecoratedProcedure[typeof utilName]
               >;
               const queryKey = getQueryKey(fullPath, input);
-              return context.setInfiniteQueryData(
+              return {
+                input,
                 queryKey,
-                updater as any,
-                ...rest,
-              );
+                updater,
+                rest,
+              };
             }
-            case 'getData': {
-              const [input] = args as Parameters<
-                AnyDecoratedProcedure['getData']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.getQueryData(queryKey);
-            }
-            case 'getInfiniteData': {
-              const [input] = args as Parameters<
-                AnyDecoratedProcedure['getInfiniteData']
-              >;
-              const queryKey = getQueryKey(fullPath, input);
-              return context.getInfiniteQueryData(queryKey);
-            }
-          }
+
+            const [input, ...rest] = args as Parameters<
+              AnyDecoratedProcedure[typeof utilName]
+            >;
+            const queryKey = getQueryKey(fullPath, input);
+            return {
+              input,
+              queryKey,
+              rest,
+            };
+          };
+
+          const { queryKey, rest, updater, input } = opts(utilName);
```

Comment:
Looks good! Can we just rename this function? We use `opts` as a variable for object parameters in a lot of places. Maybe `getOpts`?

Reply:
Agreed